### PR TITLE
Add navigation documentation

### PR DESCRIPTION
**Pull request by nirvn**
This PR adds a bit of documentation around QField's navigation features.

### CODE REVIEW

File: documentation/how-to/navigation.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+title: Navigation
+---
+
+# Navigation
+
+QField offers navigation functionalities to help orient yourself in the field and accurately reach a given destination.
+
+## Activating navigation 
+
+Navigation is enabled when a destination point has been set and positioning is active. When turned on, a set of navigation overlays - a destination marker, a navigation panel, and a navigation control button - appears over the map.
```
Comment (m-kuhn) [inline documentation/how-to/navigation.md:11]:
```suggestion
Navigation is enabled when a destination point has been set and positioning is active. When turned on, a set of navigation overlays - a destination marker, a navigation panel, and a navigation control button - appear over the map.
```

I'd say ...

Comment (m-kuhn) [reply documentation/how-to/navigation.md:11]: Could "a navigation point has been set" link to the section below?

File: documentation/how-to/navigation.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+!![](../assets/images/navigation-destination-feature-form.png)
+
+!!! note
+    When the selected feature as destination has a line or polygon geometry, a point on the geometry's surface will be used to navigate to that feature.
+
+## Auto tracking of current location and destination
+:material-tablet-android:{ .device-icon } Fieldwork
+
+QField allows for its map to automatically keep track of the currant device location and destination and re-center the map extent around those two points.
```
Comment (m-kuhn) [inline documentation/how-to/navigation.md:40]:
```suggestion
QField allows for its map to automatically keep track of the current device location and destination and re-center the map extent around those two points.
```

File: documentation/how-to/navigation.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+To activate this auto tracking feature, you can simply tap on the positioning button and the navigation control button. Both buttons should show their auto tracking mode active by having their background color turn to blue and purple.
+
+!![](../assets/images/navigation-auto-tracking.png)
+
+This can be desribed as a simple *staking mode* functionality.
```
Comment (m-kuhn) [inline documentation/how-to/navigation.md:46]:
```suggestion
This can be described as a simple *staking mode* functionality.
```

**Comment by nirvn** [comment]
@m-kuhn , review addressed.

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**